lib/enrichment/src/lib.rs

```diff
@@ -3,19 +3,26 @@ pub mod tables;
 
 #[cfg(test)]
 mod test_util;
-
-use std::collections::BTreeMap;
-
 use dyn_clone::DynClone;
+use std::collections::BTreeMap;
+use vrl_core::Value;
 
 pub use tables::{TableRegistry, TableSearch};
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub struct IndexHandle(pub usize);
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum Condition<'a> {
-    Equals { field: &'a str, value: String },
+    Equals {
+        field: &'a str,
+        value: Value,
+    },
+    BetweenDates {
```

Comment:
If I judge from the implementation correctly, this condition includes its bounds. Would add a comment here to remove any potential ambiguity